provisioning/service/src/Config/BulkEnrollmentOperation.cs

```diff
@@ -0,0 +1,101 @@
+    ///                "type":"tpm",
+    ///                "tpm":{
+    ///                    "endorsementKey":"validEndorsementKey"
+    ///                }
+    ///            },
+    ///            "iotHubHostName":"ContosoIoTHub.azure-devices.net",
+    ///            "provisioningStatus":"enabled"
+    ///        },
+    ///        {
+    ///            "registrationId":"validRegistrationId-2",
+    ///            "deviceId":"ContosoDevice-2",
+    ///            "attestation":{
+    ///                "type":"tpm",
+    ///               "tpm":{
+    ///                    "endorsementKey":"validEndorsementKey"
+    ///                }
+    ///            },
+    ///            "iotHubHostName":"ContosoIoTHub.azure-devices.net",
+    ///            "provisioningStatus":"enabled"
+    ///        }
+    ///    ]
+    /// }
+    /// </code>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iot-dps/deviceenrollment">Device Enrollment</seealso>
+    internal static class BulkEnrollmentOperation
+    {
+        private sealed class BulkOperation
+        {
+            /// <summary>
+            /// Operation mode
+            /// </summary>
+            [JsonProperty(PropertyName = "mode", Required = Required.Always)]
+            public BulkOperationMode Mode { get; set; }
+
+            /// <summary>
+            /// Enrollments for bulk operation
+            /// </summary>
+            [JsonProperty(PropertyName = "enrollments", Required = Required.Always)]
+            public IEnumerable<IndividualEnrollment> Enrollments { get; set; }
+        }
+
+        /// <summary>
+        /// Serializer
+        /// </summary>
+        /// <remarks>
+        /// Creates a <code>string</code>, whose content represents the mode and the collection of
+        ///     individualEnrollments in a JSON format.
+        /// </remarks>
+        /// <param name="mode">the <see cref="BulkOperationMode"/> that defines the single operation to do over the 
+        ///     individualEnrollments.</param>
+        /// <param name="individualEnrollments">the collection of <see cref="IndividualEnrollment"/> that contains the description 
+        ///     of each individualEnrollment.</param>
+        /// <returns>The <code>string</code> with the content of this class.</returns>
+        /// <exception cref="ArgumentNullException">if the individualEnrollments is null.</exception>
+        /// <exception cref="ArgumentException">if the individualEnrollments is invalid.</exception>
+        public static string ToJson(BulkOperationMode mode, IEnumerable<IndividualEnrollment> individualEnrollments)
+        {
+            /* SRS_BULK_OPERATION_21_001: [The toJsonElement shall throw ArgumentException if the provided collection of 
+                                            individualEnrollments is null or empty.] */
+            if (!(individualEnrollments ?? throw new ArgumentNullException("individualEnrollments")).Any())
```

Comment:
Nit: I personally prefer the long form to the new ?? operator when the code calls a member.

```C#
if (individualEnrollments == null) throw new ArgumentNullException("individualEnrollments");
if (!individualEnrollments.Any()) {...}
```

Reply:
I like the ??